### PR TITLE
[Bug Fix] NPC not breaking charm correctly

### PR DESCRIPTION
**Pull request by dencelle**
#947 and #905
fixes the issue with charm breaking and spells being cast after to cause a faction war. this removes dots to stop faction wars also.

dot removal part needs better testing to ensure it works as intended

### CODE REVIEW

**Comment by mackal** [comment]
Stripping dots sounds wrong to me.

**Comment by dencelle** [comment]
seems wrong to me too but if you don't and a dot is still on the other npc when charm is broken a war still happens.
maybe we should leave this as something you can turn on or off according to a rule variable and it being off by default?

**Comment by dencelle** [comment]
I'm not sure, according to everything I've read this problem is still
present on live as a whole and has been attempted to be fixed in a lot of
ways. This was the easiest and lightest weight way I could think of fixing
the issue, more than likely is not the best, ailia gave and idea of doing
something along the lines of a bunch of checks to see if the aggro/damage
should apply or not

**Comment by GangstaEQ** [comment]
The way it functions currently prevents most faction wars. If someone wrote the same check for whether the dot tick should provide  aggro or not that would probably be the preferred solution but how it works currently stops faction wars. +1 upvote for going to master.

File: zone/spell_effects.cpp [hunk 1]
```diff
@@ -4011,6 +4011,33 @@ void Mob::BuffFadeBySlot(int slot, bool iRecalcBonuses)
 				if (IsAIControlled())
 				{
 					// clear the hate list of the mobs
+					for (auto pMob : this->hate_list.GetHateList()) {
```
Comment (Akkadius) [inline zone/spell_effects.cpp:4014]: Rename to just `mob` please

Remove the `this->` explicits please

**Review by Akkadius — APPROVED** [review]

**Pull request merged**